contracts/votingMachines/GenesisProtocolLogic.sol

```diff
@@ -0,0 +1,652 @@
+            }
+        }
+        //calc dao bounty
+        uint daoBounty = parameters[_paramsHash].daoBountyConst.mul(averagesBoostDownstakes[proposal.organizationId]);
+        if (daoBounty < parameters[_paramsHash].minimumDaoBounty) {
+            proposal.daoBountyRemain = parameters[_paramsHash].minimumDaoBounty;
+        } else {
+            proposal.daoBountyRemain = daoBounty;
+        }
+        proposal.totalStakes = proposal.daoBountyRemain;
+        proposals[proposalId] = proposal;
+        proposals[proposalId].stakes[NO] = proposal.daoBountyRemain;//dao downstake on the proposal
+        emit NewProposal(proposalId, organizations[proposal.organizationId], NUM_OF_CHOICES, _proposer, _paramsHash);
+        return proposalId;
+    }
+
+    /**
+      * @dev expired try to execute a boosted proposal if it is expired
+      * @param _proposalId the id of the proposal
+      * @return uint expirationCallBounty the bounty amount for the expiration call
+     */
+    function expired(bytes32 _proposalId) external returns(uint expirationCallBounty) {
+        Proposal storage proposal = proposals[_proposalId];
+        require(proposal.state == ProposalState.Boosted);
+        require(_execute(_proposalId));
+        // solium-disable-next-line security/no-block-members
+        uint expirationCallBountyPercentage = (1 + now.sub(proposal.currentBoostedVotePeriodLimit + proposal.times[1]).div(15));
+        if (expirationCallBountyPercentage > 100) {
+            expirationCallBountyPercentage = 100;
+        }
+        proposal.expirationCallBountyPercentage = expirationCallBountyPercentage;
+        expirationCallBounty = expirationCallBountyPercentage.mul(proposal.stakes[YES]).div(100);
+        require(stakingToken.transfer(msg.sender, expirationCallBounty));
+        emit ExpirationCallBounty(_proposalId,msg.sender,expirationCallBounty);
+    }
+
+    /**
+     * @dev redeem a reward for a successful stake, vote or proposing.
+     * The function use a beneficiary address as a parameter (and not msg.sender) to enable
+     * users to redeem on behalf of someone else.
+     * @param _proposalId the ID of the proposal
+     * @param _beneficiary - the beneficiary address
+     * @return rewards -
+     *           [0] stakerTokenReward
+     *           [1] voterReputationReward
+     *           [2] proposerReputationReward
+     */
+    function redeem(bytes32 _proposalId,address _beneficiary) public returns (uint[3] rewards) {
+        Proposal storage proposal = proposals[_proposalId];
+        require((proposal.state == ProposalState.Executed) || (proposal.state == ProposalState.Closed),"wrong proposal state");
+        Parameters memory params = parameters[proposal.paramsHash];
+        uint lostReputation;
+        if (proposal.winningVote == YES) {
+            lostReputation = proposal.preBoostedVotes[NO];
+        } else {
+            lostReputation = proposal.preBoostedVotes[YES];
+        }
```

Comment:
This is confusing. Perhaps write `losingVote = proposalWinningVote == YES && NO || YES` ,a nd then use `preBoostedVotes[losingVote]`

Reply:
not sure I understand

Reply:
PhD in logic  ;-)